### PR TITLE
Added 0900|Sync_With_Database.md

### PR DESCRIPTION
**Pull request by mchaudhry05**
## PR Description
Created a markdown file that shows an example of how you can sync data from Homebase to Firebase and from Firebase to Homebase. 

### Testing
- [ ] added relevant test coverage
- [x] no tests needed

### Docs
- [ ] added relevant docs 
    - preview them at https://homebase.io/docs/homebase-react/{BRANCH_NAME}/overview
- [ ] updated relevant sections in the README.md
- [ ] updated relevant docstrings in index.d.ts
- [x] no docs needed

### Typescript
- [ ] added or edited relevant Typescript type declarations
- [x] no type declaration updates needed

## Merging
For maintainers.

To merge, select "Squash and Merge". Then:
  1. Make sure the top commit message follows [Angular Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines),
  2. Delete all other commit messages in the description, but keep any lines designating [co-authors](https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) so contributors will retain credit for their contributions.

### CODE REVIEW

File: docs/0900|Sync_With_Database.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+## Syncing to Database Recipie with Firebase
```
Comment (smothers) [inline docs/0900|Sync_With_Database.md:1]: Recipie -> Recipe

File: docs/0900|Sync_With_Database.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+## Syncing to Database Recipie with Firebase 
+
```
Comment (smothers) [inline docs/0900|Sync_With_Database.md:2]: The file name maps to the name of this page on the docs site. I think we should call this page Recipes so please rename the file to `0900|Recipes.md`

File: docs/0900|Sync_With_Database.md (hunk 1 above)
Comment (smothers) [inline docs/0900|Sync_With_Database.md:1]: I think this should just be `Syncing to Firebase`

File: docs/0900|Sync_With_Database.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+The example below shows a recipe for syncing Homebase with Firebase. To communicate with our backend we use the addTransactListener() method that is part of our useClient() API call. addTransactListener() takes in a function as a parameter. transactListener is the function that looks at all of our Datoms that have changed and updates them in Firebase. We also need a way to sync our backend data with Homebase. transactSilently() allows us to do exactly that and it is also part of the useClient() API call. We have a function called softTransact() which will take in a queue with all the changes that have been made in our database and use transactSilently() to notify Homebase of those changes. Debounced is used here to only improve performance as it allows us to wait for a certain number of changes to accumulate before updating Homebase.
```
Comment (smothers) [inline docs/0900|Sync_With_Database.md:3]: This looks great! I've just made a few copy edits to trim it down a bit.

The example below shows a recipe for keeping Homebase React in sync with Firebase. `client.addTransactListener(callback)` lets you listen to every local transaction and send those updates to Firebase. We also need a way to sync Firebase with Homebase React. In this example we create a namespace on Firebase for each user based on their firebase uid and listen to all changes in that namespace. `client.transactSilently(tx)` allows us save changes received from Firebase without triggering our transactListener function and sending those changes back to Firebase endlessly. 

Debounced is used here to improve performance as it allows us to batch changes that occur close together.

File: docs/0900|Sync_With_Database.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import debounce from 'lodash/debounce';
+import React from 'react';
+
+const SyncToFirebase = () => {
+    const [client] = useClient()
+    const [currentUser] = useEntity({ identity: 'currentUser' })
+    const userId = currentUser.get('uid')
+    const transactListener = React.useCallback(
+      (changedDatoms) => {
+        const cardinalityManyAttrs = new Set([])
```
Comment (smothers) [inline docs/0900|Sync_With_Database.md:17]: Let's add a comment with some examples
```js
const cardinalityManyAttrs = new Set([]) // E.g. ':project/todos' or ':user/friends'
```

File: docs/0900|Sync_With_Database.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        const localOnlyAttrs = new Set([])
```
Comment (smothers) [inline docs/0900|Sync_With_Database.md:18]:
```js
const localOnlyAttrs = new Set([]) // E.g. ':current-user/uid' these are attributes you don't want to save to Firbase, but also don't want to have to call `client.transactSilently()` everytime you change them.
```

File: docs/0900|Sync_With_Database.md (hunk 2 above)
Comment (mchaudhry05) [reply docs/0900|Sync_With_Database.md]: Changed the name to `0900|Recipes.md`!

File: docs/0900|Sync_With_Database.md (hunk 1 above)
Comment (mchaudhry05) [reply docs/0900|Sync_With_Database.md]: Changed to `Syncing to Firebase`

File: docs/0900|Sync_With_Database.md (hunk 3 above)
Comment (mchaudhry05) [reply docs/0900|Sync_With_Database.md]: Updated Description!

File: docs/0900|Sync_With_Database.md (hunk 4 above)
Comment (mchaudhry05) [reply docs/0900|Sync_With_Database.md]: Added comment

File: docs/0900|Sync_With_Database.md (hunk 5 above)
Comment (mchaudhry05) [reply docs/0900|Sync_With_Database.md]: Added comment

**Review by smothers — APPROVED** [review]
Looks great! Thanks @mchaudhry05

**Pull request merged**